Core/src/Material/Interactions.cpp

```diff
@@ -126,11 +131,13 @@ inline float deriveDeltaHalf(float qOverP, const RelativisticQuantities& rq) {
   //     d(beta*gamma)/(beta*gamma)
   return (rq.betaGamma < 10.0f) ? 0.0f : (-1.0f / qOverP);
 }
+
 }  // namespace
 
-#define ASSERT_INPUTS(mass, qOverP, q)            \
-  assert((0 < mass) and "Mass must be positive"); \
-  assert((0 < (qOverP * q)) and "Inconsistent q/p and q signs");
+#define ASSERT_INPUTS(mass, qOverP, q)                              \
+  assert((0 < mass) and "Mass must be positive");                   \
+  assert((qOverP < 0) and (0 < qOverP) and "q/p must be non-zero"); \
```

Comment:
Hmmm... my eyes glazed over this too quickly, and I think yours too. You probably meant `(qOverP < 0) or (0 < qOverP)`, which can be simplified to `(qOverP != 0)`. The only value of `qOverP` which follows the `(qOverP < 0) and (0 < qOverP)` property is NaN...

Reply:
Thanks. Just fixed it.